### PR TITLE
move husky and lintstaged setup from 'prettier' module to 'init'

### LINKED ISSUE #10701: Move husky and lintstaged setup from 'prettier' module to 'init'
Git hooks and actions on commit are not specific to prettier.
It would be nice if this was directly available on any initiated project.

An example of non-prettier usage: when you use the protobuf-backwards-compatibility-check module, it would be nice to check and update automatically the lock file when you change protobuf files.

### PR DESCRIPTION
**Pull request by fabienpuissant**
Fix #10701

### CODE REVIEW

**Comment by fabienpuissant** [comment]
It is impossible to have an empty configuration => leads to error "Configuration should not be empty"

I'm open to suggestion to handle with that. Currently I've just put a useless echo command in it

**Comment by murdos** [comment]
> It is impossible to have an empty configuration => leads to error "Configuration should not be empty"
> 
> I'm open to suggestion to handle with that. Currently I've just put a useless echo command in it

Too bad... However it works with an empty `lintstaged.cjs` file.
I suggests that we generate and empty file `lintstaged.cjs`, and then in the prettier module, we do 2 optional replacements to handle the different situations: 
1/ lintstaged.cjs is still empty => we can the add the whole `export default = { ... }`, 
2/ lintstaged.cjs has been configured meanwhile (either manually, or by another module) => we just add a new line after `export default = {`

You can find a similar situation here: https://github.com/jhipster/jhipster-lite/blob/799ff6f7b8c0b70bac48e02e52768fd4ac54e2cc/src/main/java/tech/jhipster/lite/generator/client/tools/cypress/domain/CypressModuleFactory.java#L106-L110
And the corresponding unit tests in `CypressModuleFactoryTest`.

File: src/main/resources/generator/init/package.json.mustache [hunk 1]
```diff
@@ -6,5 +6,12 @@
   "license": "UNLICENSED",
   "engines": {
     "node": ">={{nodeMajorVersion}}"
+  },
+  "scripts": {
```
[the remark this answers is not in the document]
Comment (fabienpuissant) [reply src/main/resources/generator/init/package.json.mustache:10]: Firstly I tried to use the API but as the package.json is created in this module, I had some errors (even if the addDevDependency was after the package.json insertion)

**Comment by fabienpuissant** [comment]
> > It is impossible to have an empty configuration => leads to error "Configuration should not be empty"
> > I'm open to suggestion to handle with that. Currently I've just put a useless echo command in it
> 
> Too bad... However it works with an empty `lintstaged.cjs` file. I suggests that we generate and empty file `lintstaged.cjs`, and then in the prettier module, we do 2 optional replacements to handle the different situations: 1/ lintstaged.cjs is still empty => we can the add the whole `export default = { ... }`, 2/ lintstaged.cjs has been configured meanwhile (either manually, or by another module) => we just add a new line after `export default = {`
> 
> You can find a similar situation here:
> 
> https://github.com/jhipster/jhipster-lite/blob/799ff6f7b8c0b70bac48e02e52768fd4ac54e2cc/src/main/java/tech/jhipster/lite/generator/client/tools/cypress/domain/CypressModuleFactory.java#L106-L110
> 
> 
> And the corresponding unit tests in `CypressModuleFactoryTest`.

Good idea ! Thank you

**Comment by fabienpuissant** [comment]
I needed to rename src/main/resources/generator/init/.lintstagedrc.cjs into src/main/resources/generator/init/.lintstagedrc.template.cjs because otherwise error when commiting because configuration empty

**Review by murdos — APPROVED** [review]

**Pull request merged**